test/e2e/upgrade_management_cluster_test.go

```diff
@@ -26,18 +27,19 @@ var _ = Describe("When testing cluster upgrade v1alpha5 > current [upgrade]", fu
 	})
 	capi_e2e.ClusterctlUpgradeSpec(ctx, func() capi_e2e.ClusterctlUpgradeSpecInput {
 		return capi_e2e.ClusterctlUpgradeSpecInput{
-			E2EConfig:                 e2eConfig,
-			ClusterctlConfigPath:      clusterctlConfigPath,
-			BootstrapClusterProxy:     bootstrapClusterProxy,
-			ArtifactFolder:            artifactFolder,
-			SkipCleanup:               true, // TODO(mboukhalfa): Parameterize after merging https://github.com/kubernetes-sigs/cluster-api/pull/7373
-			InitWithProvidersContract: "v1alpha4",
-			InitWithBinary:            e2eConfig.GetVariable("INIT_WITH_BINARY"),
-			PreInit:                   preInitFunc,
-			PreWaitForCluster:         preWaitForCluster,
-			PreUpgrade:                preUpgrade,
-			MgmtFlavor:                osType,
-			WorkloadFlavor:            osType,
+			E2EConfig:                   e2eConfig,
+			ClusterctlConfigPath:        clusterctlConfigPath,
+			BootstrapClusterProxy:       bootstrapClusterProxy,
+			ArtifactFolder:              artifactFolder,
+			SkipCleanup:                 skipCleanup,
```

Comment:
I'm trying to understand if this is changing something or if it will be the same. What is the default for `skipCleanup`? I'm guessing it is `false`, but we had `true` hard coded before. :thinking:

Reply:
We had true hard coded before because we where skipping  cluster deletion from bootstrap. And we had to that because capi function did allow us to move back ironic to the bootstrap which is  required for cluster deletion.

After injecting this new hook we are able to move back ironic then we delete the cluster at the end or we can skip based on the var `skipCleanup`

Reply:
Thanks for the explanation! You should put this in the description of the PR!